Documentation/SwiftlyDocs.docc/swift-version.md

```diff
@@ -0,0 +1,97 @@
+
+swiftly is in the position of being able to install toolchains and switch between them automatically. If there is a standard way to write down the default version in the project's repository then it has the information that it needs to both install and switch to the recommended toolchain on behalf of the developer, and the CI system.
+
+This is the purpose of the `.swift-version` file. It installs the recommended toolchain that should get you a reliable development environment, and CI environment given the current state of the project code. The format is very simple, just a single line with the full toolchain version:
+
+```
+6.1.1
+```
+
+That's all there is to it. It can be easily read/written by tools, including swiftly, because it's very simple. In the worst case, it is just another form of documentation, but standardized in terms of location and format. The reliability increases as more workflows begin using it. Developers use this toolchain to advance the code and establish a recommended version. CI systems can verify and block any changes that break with this version. Finally, developers can switch between projects that make use of it much more fluidly and easily.
+
+swiftly is contributing to this effort by reading this file automatically. When switching toolchains it seeds them when you `swiftly install` or `swiftly use` a different toolchain.
+
+### Other versions: Swift Tools
+
+There are other sorts of versions in Swift. For a typical SwiftPM package, there's the "Swift Tools Version" at the top of the `Package.swift` file:
+
+```
+//swift-tools-version: 5.10
+
+Package(
+   name: "foo"
+)
+```
+
+From the [SwiftPM documentation](https://docs.swift.org/package-manager/PackageDescription/PackageDescription.html#about-the-swift-tools-version) the first purpose of this version is to declare the version of the Package.swift file's contents so that it can be processed using the correct version of the package description structure. Another purpose is to set the minimum version of the "Swift tools" that are needed to use the package.
+
+From this first purpose, this has not so much to do with the toolchain version, as it does with the need to understand the package description that's inside the file. Newer toolchains will fall back to this version of the description to get the package description. It's also in the best interest of a package to use an older version for this so that downstream packages can continue to use older toolchains.
+
+In terms of the second purpose this acts as a lower-bound of the swift tools for reasons stated above. There are serious tensions between using this as the exact recommended version because there are additional concerns with developing, and testing this package that are not shared by consumers of it, such as features used in testing, formatting, and IDE integration.
```

Comment:
```suggestion
In terms of the second purpose this acts as a lower-bound of the swift tools for reasons stated above. There are tensions between using this as an exact version because there of concerns with developing and testing that aren't shared by consumers, such as features used in testing, formatting, and IDE integration.
```